xls/dslx/ir_convert/ir_converter_options_flags_test.cc

```diff
@@ -0,0 +1,64 @@
+// Copyright 2025 The XLS Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "xls/dslx/ir_convert/ir_converter_options_flags.h"
+
+#include "absl/cleanup/cleanup.h"
+#include "absl/flags/flag.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "xls/common/status/matchers.h"
+#include "xls/dslx/warning_kind.h"
+
+ABSL_DECLARE_FLAG(std::optional<std::string>, enable_warnings);
+ABSL_DECLARE_FLAG(std::optional<std::string>, disable_warnings);
+ABSL_DECLARE_FLAG(bool, warnings_as_errors);
+
+namespace xls {
+
+TEST(IrConverterOptionsFlagsTest, WarningOptionsNoFlagSettings) {
+  XLS_ASSERT_OK_AND_ASSIGN(IrConverterOptionsFlagsProto options,
+                           GetIrConverterOptionsFlagsProto());
+  EXPECT_TRUE(options.enable_warnings().empty());
+  EXPECT_TRUE(options.disable_warnings().empty());
+  EXPECT_TRUE(options.has_warnings_as_errors());
+  EXPECT_EQ(options.warnings_as_errors(), true);
+}
+
+TEST(IrConverterOptionsFlagsTest,
+     WarningOptionsBothEnableAndDisableSetsMultiEntry) {
+  // Validate the default value we're going to reset this flag to.
+  EXPECT_EQ(absl::GetFlag(FLAGS_warnings_as_errors), true);
```

Comment:
I think you should check all the flags, right? It might make sense for these to be `ASSERT_EQ()` too.